pkg/secrets-store/nodeserver.go

```diff
@@ -282,22 +268,26 @@ func (ns *nodeServer) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpu
 	targetPath := req.GetTargetPath()
 	volumeID := req.GetVolumeId()
 
-	// check if target path is mount point
-	notMnt, err := mount.New("").IsLikelyNotMountPoint(targetPath)
-	// if not mount point then no need to unmount
-	if (err != nil && os.IsNotExist(err)) || notMnt {
-		log.Infof("target path %s is not likely a mount point, notMnt %v", targetPath, notMnt)
-		return &csi.NodeUnpublishVolumeResponse{}, nil
-	}
-	if err != nil && !os.IsNotExist(err) {
-		return nil, err
+	if runtime.GOOS == "windows" {
+		files, err := filepath.Glob(filepath.Join(targetPath, "*"))
+		if err != nil {
+			log.Errorf("failed to list dir for target path %s, err: %v", targetPath, err)
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+		for _, file := range files {
+			err = os.RemoveAll(file)
+			if err != nil {
+				log.Errorf("failed to remove file %s, err: %v", file, err)
+				return nil, status.Error(codes.Internal, err.Error())
+			}
+		}
 	}
-	// Unmounting the target
-	err = mount.New("").Unmount(targetPath)
+	err := mount.CleanupMountPoint(targetPath, ns.mounter, false)
```

Comment:
Can you double check that `extensiveMountPointCheck` should be set to `false` for Windows, which uses `IsLikelyNotMountPoint`
https://github.com/kubernetes/utils/blob/master/mount/mount_helper_common.go#L26-L28

Reply:
I've tested with `extensiveMountPointCheck` true too, but the List mountpoints interface is not implemented for windows which means it always returns not mount. I'll double check again.

Reply:
Validated this! `IsMountPoint` done for extensive check calls list mount interface (which isn't implemented for windows). So it always returns not mnt.